asllib/doc/ASLTypingReference.tex

```diff
@@ -721,156 +1034,25 @@ \subsection{Subsumption Testing}
-
-We use the shorthand notation $\unconstrainedinteger$ to denote the unconstrained integer type: $\TInt(\langle\rangle)$.
-
-Note that throughout this document we use $\tty$ to denote a type variable, which should not be confused with the abstract syntax variable $\ty$.
-
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-\chapter{Reading guide}
-%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-
-The definition of each \texttt{annotation\_<label>} function is given by a number of
-rules, which follow the possible shapes the \texttt{label} can have. For
-example, an expression can be a literal, or a binary operator, amongst other
-things. Each of those has its own evaluation rule: TypingRule.ELit in
-Section~\secref{TypingRule.ELit} and
-Typing.Binop in Section~\secref{TypingRule.Binop}, respectively.
-
-Each rule is presented using the following template:
-\begin{itemize}
-\item a Prose paragraph gives the rule in English, and corresponds as much as possible to the code of the reference implementation ASLRef given at
-\href{https://github.com/herd/herdtools7//tree/master/asllib}{/herdtools7/asllib};
-\item one or several Example paragraphs, which as much as possible are also given as regression tests in
-\href{https://github.com/herd/herdtools7//tree/master/asllib/tests/ASLTypingReference.t}{/herdtools7/asllib/tests/ASLTypingReference.t};
-\ifcode
-\item a Code paragraph which gives a verbatim of the corresponding OCaml implementation in the type-checker of ASLRef
-\href{https://github.com/herd/herdtools7//tree/master/asllib/Typing.ml}{/herdtools7/asllib/Typing.ml};
-\fi
-\item Formal paragraphs which give formal definitions of the rule.
-\item Comments paragraphs which may refer to one or more rules from the Language Reference Manual~\cite{LRM}
-      and may add more information.
-\end{itemize}
+  \item[False Negative:] the symbolic reasoning is unable to decide.
+\end{description}
+
+In other words, $\symsubsumes(\tenv, \vt, \vs)$ errs on the \emph{safe side} ---
+it never answers $\True$ when the real answer is $\False$, which would (undesirably)
+determine the following statement as well-typed:
+\begin{center}
+  \verb|var x : integer{1,2} = integer;|
+\end{center}
+
+A sound but trivial subsumption test is one that always returns $\False$.
+However, that would make many useful examples, such as the one above,
+be considered as not well-typed. Indeed, it has the maximal set of false negatives.
+Reducing the set of false negatives requires stronger symbolic reasoning algorithms,
+which inevitably leads to higher computational complexity.
+%
+The symbolic subsumption test in \chapref{symbolicsubsumptiontesting}
+attempts to accept a large enough set of true positives, based on empirical trial and error,
+while maintaining the computational complexity of the symbolic reasoning acceptable.
```

Comment:
acceptable -> relatively low

also, please could you expand on which trade-offs you've had to make?

Reply:
Moved to a PR covering the symbolic subsumption testing.